content/doc/book/pipeline/syntax.adoc

```diff
@@ -0,0 +1,851 @@
+
+[[when-example]]
+===== Example
+
+[pipeline]
+----
+// Declarative //
+pipeline {
+    agent any
+    stages {
+        stage('Example Build') {
+            steps {
+                echo 'Hello World'
+            }
+        }
+        stage('Example Deploy') {
+            when {
+                branch '*/production'
+            }
+            steps {
+                echo 'Deploying'
+            }
+        }
+    }
+}
+----
+
+[[declarative-steps]]
+=== Steps
+
+Declarative Pipelines may use all the available steps documented in the
+link:/doc/pipeline/steps[Pipeline Steps reference],
+which contains a comprehensive list of steps, with the addition of the steps
+defined below which are *only-supported* in Declarative Pipeline.
+
+==== script
+
+The `script` step takes a block of <<scripted-pipeline>> and executes that in
+the Declarative Pipeline. For most use-cases, the `script` step should be
+unnecessary in Declarative Pipelines, but it can provide a useful "escape
+hatch." `script` blocks of non-trivial size and/or complexity should be moved
+into <<shared-libraries#, Shared Libraries>> instead.
+
+[[script-example]]
+===== Example
+
+[pipeline]
+----
+// Declarative //
+pipeline {
+    agent any
+    stages {
+        stage('Example') {
+            steps {
+                echo 'Hello World'
+
+                script {
+                    def browsers = ['chrome', 'firefox']
+                    for (int i = 0; i < browsers.size(); ++i) {
+                        echo "Testing the ${browsers[i]} browser"
+                    }
+                }
+            }
+        }
+    }
+}
+----
+
+[role=syntax]
+== Scripted Pipeline
+
+Scripted Pipeline, like <<declarative-pipeline>>, is built on top of the
+underlying Pipeline sub-system. Unlike Declarative, Scripted Pipeline is
+effectively a general purpose DSL
+footnoteref:[dsl,Domain-specific Language]
+built with
+link:http://groovy-lang.org/syntax.html[Groovy].
+Most functionality provided by the Groovy language is made available to users
+of Scripted Pipeline, which means it can be a very expressive and flexible tool
+with which one can author continuous delivery pipelines.
+
+
+=== Flow Control
+
+Scripted Pipeline is serially executed from the top of a `Jenkinsfile`
+downwards, like most traditional scripts in Groovy or other languages.
+Providing flow control therefore rests on Groovy expressions, such as the
+`if/else if/else` conditionals, for example:
+
+[pipeline]
+----
+// Scripted //
+node {
+    stage('Example') {
+        if (env.BRANCH_NAME == 'master') {
+            echo 'I only execute on the master branch'
+        }
+    }
+}
+----
+
+Another way Scripted Pipeline flow control can be managed is with Groovy's
+exception handling support. When <<scripted-steps>> fail for whatever reason
+they throw an exception.  Meaning handling on-error behaviors must make use of
+the `try/catch/finally` blocks in Groovy, for example:
```

Comment:
"Meaning handling on-error behaviours" is a wee bit tough to parse.

Maybe "This means that handling errors"?